### PR TITLE
Update project visibility flags

### LINKED ISSUE #941: Project visibility fields
For the interface described in zooniverse/Panoptes-Front-End#240, projects will need some additional fields:

- `private` (Already exists)
- `beta_requested` (Currently just `beta`, which is vague)
- `beta_approved` (Only writable by Zoo admin)
- `launch_requested`
- `launch_approved` (Only writable by Zoo admin)
- `live` (Exists; when changed to false, should also toggle `{beta,launch}_{requested,approved}` to false)

### PR DESCRIPTION
**Pull request by edpaget**
Make sure they're in line with zooniverse/Panoptes-Front-End#243

Closes #941

### CODE REVIEW

File: app/controllers/api/v1/projects_controller.rb [hunk 1]
```diff
@@ -75,18 +75,22 @@ def content_from_params(ps)
...
   end
 
   def build_update_hash(update_params, id)
-    admin_allowed(:approved, :redirect)
+    admin_allowed(:beta_approved, :launch_approved, :redirect)
     content_update = content_from_params(update_params)
     unless content_update.blank?
       Project.find(id).primary_content.update!(content_update)
     end
+    if update_params[:live] == false
```
Comment (camallen) [inline app/controllers/api/v1/projects_controller.rb]: I think we'll need to add a little more state checking / modification here to ensure the projects switch between states correctly. A state machine like library helper class perhaps?
E.g. a project is 
``` JSON
{ "live": true, "beta_requested": true, "beta_approved": false, "launch_requested": false, "launch_approved": false }
```

and then an api request comes in to just set `{ launch_requested: true }`. The project states would then be 
``` JSON
{ "live": true, "beta_requested": true, "beta_approved": false, "launch_requested": true, "launch_approved": false }
```

Comment (edpaget) [inline app/controllers/api/v1/projects_controller.rb]: I'm not sure that's necessary. Project states that aren't 'live' are not used to change any behaviour other than what's shown where on the front-end. What would the problem be in having a project with both the requested flags set?

Comment (camallen) [inline app/controllers/api/v1/projects_controller.rb]: So i guess my only concern is having clearly defined states that we use in review to understand a projects progression from `initial` -> `beta` -> `launch` and how we filter these projects to zoo team users. 

Having the above scenario where both `beta_requested: true, launch_requested: true` and then an owner unlocks their project via `live: false` will result in both the requested fields set. This makes it harder to track what state the project is actually in.

**Comment by camallen** [comment]
Also we'll need to add the check that only owner's can remove the lock state of a project from [here](https://github.com/zooniverse/Panoptes-Front-End/issues/240#issuecomment-98984793).
>Lock: If Lock = F, any collaborator can make changes. If Lock = T, only the project owner can make changes.

**Comment by edpaget** [comment]
Yeah I disagree 100% with that. Owners and Collaborators should be treated the same since they are everywhere else. I'm not sure why that particular variable needs to be different.

**Comment by camallen** [comment]
@aliburchard any reason in particular for only allowing the owner to make changes to a live / locked project?

File: app/controllers/api/v1/projects_controller.rb (hunk 1 above)
Comment (edpaget) [inline app/controllers/api/v1/projects_controller.rb]: I'm not sure I understand what's unclear about the the project state in that case. The state is that the project isn't live, and the user has requested review for both launching and beta. All of that information can be presented in whatever interface we have for reviewing projects and the reviewer can determine what needs to happen next. 

If we need to keep track of the history of what's happened to project (ie that is has had a beta period) that seems like an entirely separate issue.

**Comment by camallen** [comment]
Silence is deafening, merging this and we'll review later.

**Pull request merged**